src/main/java/org/fit/cssbox/layout/VisualContext.java

```diff
@@ -634,14 +634,16 @@ else if (src instanceof RuleFontFace.SourceURL)
                             String regName = FontDecoder.findRegisteredFont(url);
                             if (regName == null)
                             {
-                                DocumentSource imgsrc = viewport.getConfig().createDocumentSource(url);
-                                Font newFont = FontDecoder.decodeFont(imgsrc, format);
-                                if (GraphicsEnvironment.getLocalGraphicsEnvironment().registerFont(newFont))
-                                    log.debug("Registered font: {}", newFont.getFontName());
-                                else
-                                    log.debug("Failed to register font: {} (not fatal, probably already existing)", newFont.getFontName());
-                                regName = newFont.getFontName();
-                                FontDecoder.registerFont(url, regName);
+                                try (DocumentSource imgsrc = viewport.getConfig().createDocumentSource(url))
```

Comment:
As we can read in java doc for InputStream obtained from url connection, it should be closed as fast as possible. So we should do it after we load a font.